src/main/java/vg/civcraft/mc/civmodcore/util/TextUtil.java

```diff
@@ -0,0 +1,253 @@
+package vg.civcraft.mc.civmodcore.util;
+
+import java.util.*;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import org.bukkit.ChatColor;
+import org.bukkit.entity.Player;
+
+public class TextUtil
+{	
+	private static Map<String, String> tags = new HashMap<String, String>();
+	
+	static {
+		tags.put("g", TextUtil.parseColor("<lime>"));		// good
+		tags.put("b", TextUtil.parseColor("<rose>"));		// bad
+		tags.put("i", TextUtil.parseColor("<yellow>"));		// info
+		tags.put("a", TextUtil.parseColor("<gold>"));		// art
+		tags.put("l", TextUtil.parseColor("<green>"));		// logo
+		tags.put("n", TextUtil.parseColor("<silver>"));		// notice
+		tags.put("h", TextUtil.parseColor("<pink>"));		// highlight
+		tags.put("c", TextUtil.parseColor("<aqua>"));		// parameter
+		tags.put("p", TextUtil.parseColor("<teal>"));		// parameter
+		tags.put("w", TextUtil.parseColor("<white>"));		// parameter
+		tags.put("lp", TextUtil.parseColor("<lpurple>"));
+	}
+	
+	public static Map<String, String> getTags() {
+		return tags;
+	}
+	
+	// -------------------------------------------- //
+	// Top-level parsing functions.
+	// -------------------------------------------- //
+	
+	public static String parse(String str, Object... args) {
+		return String.format(parse(str), args);
+	}
+	
+	public static String parse(String str) {
+		return parseTags(parseColor(str));
+	}
+	
+	// -------------------------------------------- //
+	// Tag parsing
+	// -------------------------------------------- //
+	
+	public static String parseTags(String str) {
+		return replaceTags(str, tags);
+	}
+	
+	public static final transient Pattern patternTag = Pattern.compile("<([a-zA-Z0-9_]*)>");
+	public static String replaceTags(String str, Map<String, String> tags) {
+		StringBuffer ret = new StringBuffer();
+		Matcher matcher = patternTag.matcher(str);
+		while (matcher.find())
+		{
+			String tag = matcher.group(1);
+			String repl = tags.get(tag);
+			if (repl == null)
+			{
+				matcher.appendReplacement(ret, "<"+tag+">");
+			}
+			else
+			{
+				matcher.appendReplacement(ret, repl);
+			}
+		}
+		matcher.appendTail(ret);
+		return ret.toString();
+	}
+	
+	// -------------------------------------------- //
+	// Color parsing
+	// -------------------------------------------- //
+	
+	public static String parseColor(String string) {
+		string = parseColorAmp(string);
+		string = parseColorAcc(string);
+		string = parseColorTags(string);
+		return string;
+	}
+	
+	public static String parseColorAmp(String string) {
+		string = string.replaceAll("(§([a-z0-9]))", "\u00A7$2");
+		string = string.replaceAll("(&([a-z0-9]))", "\u00A7$2");
+		string = string.replace("&&", "&");
+		return string;
+	}
+	
+	public static String parseColorAcc(String string) {
+		return string.replace("`e", "")
```

Comment:
Ah so this is a custom encoding? I admit to not looking too closely at it. Minecraft has a "standard" encoding that's compatible with book data entry: http://minecraft.gamepedia.com/Formatting_codes and this encoding is translated by the method I mentioned, where the `char` param is the prefix prior to the Minecraft color code.